gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/uma/resources/endpoint/ResourceRegistrationEndpoint.java

```diff
@@ -72,15 +72,15 @@ public void handle(RoutingContext context) {
         Client client = context.get(ConstantKeys.CLIENT_CONTEXT_KEY);
         subjectManager.findUserIdBySub(accessToken)
                 .switchIfEmpty(Single.error(() -> new UserNotFoundException(accessToken.getSub())))
-                .flatMap(userId -> this.resourceService.listByDomainAndClientAndUser(domain.getId(), client.getId(), userId)
-                .map(Resource::getId)
-                .collect(JsonArray::new, JsonArray::add))
+                .flatMap(userId -> this.resourceService.listByDomainAndClientAndUser(domain.getId(), client.getId(), userId.id()/*todo degraded mode: should we use full id here as well?*/)
```

Comment:
id() is fine. This entity is currently managed only in the CP (not in the DP) so it is not possible to create it (or access it) without a valid user internalId